interpreter/cling/lib/Interpreter/NullDerefProtectionTransformer.cpp

```diff
@@ -49,29 +50,33 @@ namespace cling {
     PointerCheckInjector(Sema& S) : m_Sema(S), m_Context(S.getASTContext()),
     m_clingthrowIfInvalidPointerCache(0) {}
 
-    void VisitStmt(Stmt* S) {
+    bool VisitStmt(Stmt* S) {
       for (auto child: S->children()) {
         if (child)
-          Visit(child);
+            VisitStmt(child);
       }
+      return true;
     }
 
-    void VisitUnaryOperator(UnaryOperator* UnOp) {
-      Visit(UnOp->getSubExpr());
-      if (UnOp->getOpcode() == UO_Deref) {
-        UnOp->setSubExpr(SynthesizeCheck(UnOp->getSubExpr()));
-      }
+    bool VisitUnaryOperator(UnaryOperator* UnOp) {
+      VisitStmt(UnOp->getSubExpr());
```

Comment:
`getSubExpr()` is not entirely free and you use it up to three times. Can you store it in a local?